django/db/models/sql/query.py

```diff
@@ -606,8 +613,11 @@ def deferred_to_data(self, target, callback):
                 if not is_reverse_o2o(source):
                     must_include[old_model].add(source)
                 add_to_dict(must_include, cur_model, opts.pk)
-            field, model, _, _ = opts.get_field_by_name(parts[-1])
-            if model is None:
+            field = opts.get_field(parts[-1])
+            is_reverse_object = field.auto_created and not field.concrete
```

Comment:
Again, it would be be great if we could handle non-auto-created `one_to_many` related fields. Not sure what this code does, but would it work to do:
`is_reverse_object = field.related_model and not field.concrete`?